### ISSUE TITLE
Focus Fire usage on Cooldown could get dps increase?

### ISSUE THREAD

**Issue by navv1234** [simulationcraft/simc#2386]
Originally reported on Google Code with ID 2387

```
hi,
i have a question and asked it already in a hunter forum at icy veins, but wasn't really
convinced with the answers especially after i run a few crude tests to get crude numbers
but which should point to the right direction. 

Topic in Icy veins forum http://www.icy-veins.com/forums/topic/9474-use-focus-fire-on-cooldown/
(i think there it might be better to read^^) 

The copied text will be below. If you don't want to follow some link of a stranger^^.
After you have read here would be my question .. are they right ? Is this already inside
the calculations? For my understanding it seems that "focus fire" should be used on
cd maybe even if it is only 1 stack, maybe only if at least 2 stacks.

(First)
In your Beast Mastery Guide http://www.icy-veins...downs-abilities there stand you
should use Focus Fire with 5 stacks and a bit lower some advice about using it during
bestial wrath. A information you find everywhere.

But i think it doesn't matter how much stacks you have .. use it on the spot. Because
the Pet gets only 4% haste from 1 stack of Frenzy but 6% haste from you if you have
1 stack of Focus Fire. Because Hunter Haste = Pet Haste

So even if you only watch the Pet DPS it would increase through this additional 2%
haste. But now also the hunter gets 6% Haste and 5% AP.

I got a few numbers on a short testing:

FrenzyStacks >> FocusFireStacks >> Pet Attack Speed
0 >> 0 >> 1,69 (i have 7,73% Haste so my pet aswell)
1 >> 0 >> 1,62
0 >> 1 >> 1,59
1 >> 1 >> 1,53
5 >> 0 >> 1,41
0 >> 5 >> 1,30 (so i even would say USE IT DEFINITLY before bestial wrath)

You would loose a bit dps if your Pet Frenzy stacks get around 3 or higher before your
Focus Fire (with 1 stacks) runs out. No maths to proof only a feeling. Maybe some math
crack can check at which stacks you should use focus fire that you can get the most
use out of it .. i suggest 2 stacks maybe already 1 stack.

So i hope i didn't miss a important point that makes my suggestions incorrect. Would
already be incorrect if the point that "hunter haste = pet haste" is false .. but my
windows ingame show it me that way.

(Second)
hmm okay Global Cooldown.

But it wouldn't only a few times be 1 stacks. Okay would be all 20 seconds a global
cd but every use of it gets you a damage increase for the 20seconds.
How much worth is a global cd ? Let's say how much damage increase should it be to
be used instead of an arkanshot?

i'm missing a bit the time where someone was doing the maths and didn't use a tool
where he may don't know the used mechanics for it. For example the 1 Stacks DPS Ranking
seems to be an output if it would be EVERY time only 1 stack .. and that wouldn't be
the case ..

*edit
i'm sorry if you exactly know the maths behind the DPS Ranking tool you used ^^


(Third)
Because even a 6% HunterHaste, 2% Pethaste, 5% AP increase is a huge Buff and that
would be only 1 Stack of Focus Fire, wouldn't you say so ? And it would only use a
CD for Arcanshot (our least priotorized globalcd)

Or lets say it is for 20 seconds:
12% Hunterhaste, 4% Pethaste, 10% AP vs 1 Arcanshot
18% Hunterhaste, 6% Pethaste, 15% AP vs 1 Arcanshot
24% Hunterhaste, 8% Pethaste, 20% AP vs 1 Arcanshot
30% Hunterhaste, 10% Pethaste, 25% AP vs 1 Arcanshot
gets your pet also a bit focus..

I would maybe choose from 2 stacks on and only if i need burst in the next few seconds
i would wait and hope for more stacks


(Fourth)



Hmm i have tried a little bit, all numbers are for my hunter without trinkets and ring
and please be aware that they are far from accurate the numbers are only going in the
direction to get a roughly impression:

arcane per shot >> 7794
auto per shot >> 3627
cobra per shot >> 4623
kill command >> non crit 10893

in 20 seconds approx 8 autoshot, 5 arcane (only pick a number what could be realistic
and not too much), 5 cobrashot, 2x kill command

Autoshots:
for 1 stack focus fire an Autoshot gets something around 140 damage more and i would
add 6% more for the haste buff so for autoshot would be
8x 3627 = 29016
8x (3627+140) x 1,06 = 31944,16
---------
31944,16 - 29016 = 2928 damage in 20 seconds

Cobrashots:
for 1 stack focus fire an cobrashot gets around 124 damage also cast time reduced by
6% so 6% more cobrashots? so 6% damage increase
5 x 4263 = 21315
5 x (4263+124) x 1,06 = 23251,1
----------
23251,1-21315 = 1936 damage in 20 seconds

Arkanshot
+ around 2,8% damage increase for the ap 7794x1,028
5x 7794 = 38970
5x 7794x1,028= 40061
----------
40061 - 38970 = 1091 damage in 20 seconds

kill command
seems to get around 5% damage increase for 5% ap in my case
2x 10893 = 21786
2x 10893x 1,05 = 22875
----------
22875 - 21786 = 1089 damage in 20 seconds
____________

2928+1936+1091+1089= 7044 more damage during the next 20 seconds for 1 stack
vs
1 arkanshot = 7794

2 stacks should net you around 14k more

and the pet also gets the buffs
but the arcan also get you with 20% chance a procc for pet attacks

so in conclusion i would say .. 1 stack = 1 arcane shot, 2 stacks > arcane shot...

I would love it if someone who knows the theory behind it to take a look and did a
longer test.

```

Reported by `BeZenichowski` on 2015-01-15 16:39:59
